services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts

```diff
@@ -0,0 +1,420 @@
+    registration,
+    oldRegistration,
+    isRetry,
+    status,
+    errorText: errorMessage,
+  }: ProcessTransactionResultInput): Promise<void> {
+    const resultTransaction = await this.createTransaction({
+      amount: calculatedTranserAmountInMajorUnit,
+      registration: registration,
+      financialServiceProviderId: financialServiceProviderId,
+      programId: programId,
+      paymentNumber: paymentNumber,
+      userId: userId,
+      status: status,
+      errorMessage: errorMessage,
+    });
+
+    await this.latestTransactionRepository.insertOrUpdateFromTransaction(
+      resultTransaction,
+    );
+
+    if (!isRetry) {
+      await this.updatePaymentCountAndStatusInRegistration(
+        registration,
+        programId,
+      );
+      // Added this check to avoid a bit of processing time if the status is the same
+      if (
+        oldRegistration.registrationStatus !== registration.registrationStatus
+      ) {
+        await this.eventsService.log(
+          {
+            id: oldRegistration.id,
+            status: oldRegistration.registrationStatus ?? undefined,
+          },
+          {
+            id: registration.id,
+            status: registration.registrationStatus ?? undefined,
+          },
+          {
+            registrationAttributes: ['status'],
+          },
+        );
+      }
+    }
+  }
+
+  private async addMessageJobToQueue({
+    registration,
+    userId,
+    message,
+    bulksize,
+  }: {
+    registration: RegistrationEntity | Omit<RegistrationViewEntity, 'data'>;
+    userId: number;
+    message?: string;
+    messageTemplateKey?: string;
+    bulksize?: number;
+  }): Promise<void> {
+    await this.queueMessageService.addMessageToQueue({
+      registration: registration,
+      message: message,
+      messageContentType: MessageContentType.payment,
+      messageProcessType:
+        MessageProcessTypeExtension.smsOrWhatsappTemplateGeneric,
+      bulksize: bulksize,
+      userId: userId,
+    });
+  }
+
+  private async createTransaction({
+    amount, // transaction entity are always in major unit
+    registration,
+    financialServiceProviderId,
+    programId,
+    paymentNumber,
+    userId,
+    status,
+    errorMessage,
+  }: {
+    amount: number;
+    registration: RegistrationEntity;
+    financialServiceProviderId: number;
+    programId: number;
+    paymentNumber: number;
+    userId: number;
+    status: StatusEnum;
+    errorMessage?: string;
+  }) {
+    const transaction = new TransactionEntity();
+    transaction.amount = amount;
+    transaction.created = new Date();
+    transaction.registration = registration;
+    transaction.financialServiceProviderId = financialServiceProviderId;
+    transaction.programId = programId;
+    transaction.payment = paymentNumber;
+    transaction.userId = userId;
+    transaction.status = status;
+    transaction.transactionStep = 1;
+    transaction.errorMessage = errorMessage ?? null;
```

Comment:
Is this not possible?

```suggestion
    const transaction = new TransactionEntity({
        amount,
        created:  new Date(),
        registration,
        financialServiceProviderId,
        programId,
        payment,
        userId,
        status,
        transactionStep:  1,
        errorMessage: errorMessage ?? null,
    });
```

Reply:
AFAIR not possible indeed. But I am not sure anymore: was a long time ago. We could try again.